### PR TITLE
Refactoring Options

### PR DESCRIPTION
**Pull request by chaosmail**
Refactoring + Tests

### CODE REVIEW

File: src/utils/Options.ts [hunk 1]
```diff
@@ -18,84 +37,131 @@ module n3Charts.Utils {
...
         throw new TypeError('Unknown series type: ' + type);
       }
 
-      return this.series.filter((s) =>
-        s.type.indexOf(type) > -1
-      );
+      return this.series.filter((series) => series.type.indexOf(type) !== -1);
+    }
+
+    public static uuid(): string {
```
Comment (lorem--ipsum) [inline src/utils/Options.ts]: Do we use this somewhere ? I don't remember why we added this...

**Comment by lorem--ipsum** [comment]
It's pretty good, however I'm not a big fan of testing whether subfunctions have been called or not (I'm thinking of `parseSeries`, etc.). It might make refactoring difficult, and it's not really what we're interesting in. Maybe just testing the functions output would be more interesting ?

File: src/utils/Options.ts (hunk 1 above)
Comment (chaosmail) [inline src/utils/Options.ts]: I added this function to create an ID for series, where no ID is defined. Not sure if this is a good idea, but I think it's nicer than to throw an error if no ID is defined.

**Comment by chaosmail** [comment]
yeah I thought about this. this is what I came up with:
* the parse function needs to call all parse* functions
* every parse* function needs to merge default arguments with arguments and call a sanitize* function
* the sanitize* function sanitizes the options

well, I would love to remove the sanitze* functions and put them into the parse* functions, however then I cannot test if the arguments get merged properly with the default arguments.

also, parse* functions return the return value of sanitize* functions. If we test the return values we would have some redundant tests.

Let me know what you think, I definitley agree to better arrange functions and tests!

**Comment by lorem--ipsum** [comment]
Hmmmm that's true... let's keep this and wee if that's annoying when we need to maintain the code.

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Pull request merged**